src/authorities/MetricsAuthority.ts

```diff
@@ -0,0 +1,82 @@
+
+        // eslint-disable-next-line @typescript-eslint/no-misused-promises
+        this.metricsTimer = setInterval(async () => {
+            await this.gatherRedisMetrics();
+        }, this.metricsTime);
+
+        MetricsAuthority.logger.debug('Created MetricsAuthority timers');
+    }
+
+    public stop(): void {
+        MetricsAuthority.logger.debug('Clearing MetricsAuthority timers');
+
+        if (this.metricsTimer) {
+            clearInterval(this.metricsTimer);
+        }
+    }
+
+    // Always set these metrics up on runtime otherwise it makes graphs and alerts go wonky
+    private setDefaults() {
+        this.metricsHandler.increaseCounter(METRICS_NAMES.BROKER_COUNT, {broker: 'character', result: METRIC_VALUES.CACHE_MISS});
+        this.metricsHandler.increaseCounter(METRICS_NAMES.CACHE_COUNT, {type: 'item', result: METRIC_VALUES.CACHE_MISS});
+        this.metricsHandler.increaseCounter(METRICS_NAMES.BROKER_COUNT, {broker: 'item', result: METRIC_VALUES.CACHE_MISS});
+        this.metricsHandler.increaseCounter(METRICS_NAMES.CACHE_COUNT, {type: 'item', result: METRIC_VALUES.CACHE_MISS});
+        this.metricsHandler.increaseCounter(METRICS_NAMES.BROKER_COUNT, {broker: 'facility_data', result: METRIC_VALUES.CACHE_MISS});
+        this.metricsHandler.increaseCounter(METRICS_NAMES.CACHE_COUNT, {type: 'facility_data', result: METRIC_VALUES.CACHE_MISS});
+    }
+
+    private async gatherRedisMetrics() {
+        // Total up the number of keys currently in various caches
+        const promises = [
+            await this.cacheClient.keys('cache:character:*'),
+            await this.cacheClient.keys(`cache:item:${this.censusEnvironment}:*`),
+            await this.cacheClient.keys(`cache:facilityData:${this.censusEnvironment}:*`),
+            await this.cacheClient.smembers(`unknownItems:${this.censusEnvironment}`),
+            await this.cacheClient.smembers(`unknownFacilities:${this.censusEnvironment}`),
+            await this.cacheClient.keys('characterPresence:*'),
+            await this.cacheClient.keys('outfitParticipants:*'),
+        ];
+
+        await Promise.all(promises).then((results) => {
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[0].length, {type: 'cache_character'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[1].length, {type: 'cache_item'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[2].length, {type: 'cache_facility_data'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[3].length, {type: 'unknown_items'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[4].length, {type: 'unknown_facilities'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[5].length, {type: 'character_presence'});
+            this.metricsHandler.setGauge(METRICS_NAMES.CACHE_GAUGE, results[6].length, {type: 'outfit_participants'});
+        });
```

Comment:
I think this is unnecessary. Defaults are not needed as Prometheus is able to infer this by itself. As for the Redis stuff maybe consider select? To the Gauge configuration add a collect method that runs a query on Redis. One of the considerations is that metrics are suppose to be simple addition, hence there shouldn't be a need for these kinds of complexities.

Example gauge provider:
```ts
{
  provider: getToken('tralala_total'),
  useFactory: (redis: Redis) => getOrCreateMetric('Gauge', {
    name: 'tralala_total',
    help: 'Lorem ipsum',
    collect: () =>{
      this.set(redis.someQueryThatReturnsNumberWrappedInPromise());
    }
},
inject: [Redis]

Reply:
Removed `setDefault()` 2bb13070dc04d2795893f0154526ec3b8207b3b9

Reply:
In terms of the collect method, can this be achieved using labels? Also not quite sure where this provider is meant to be injected etc

Reply:
I am pretty sure that `this.set()` in collect will accept labels if that is what you mean. The provider needs to be registered in any module or its own. Wherever you do it you just import redis or any other modules that hold the providers that you infer the metrics from, then inject the providers and build the metric. The metric itself doesn't have to be exported or injected anywhere, whenever the metrics endpoint will be scraped the collect method will be called first and set the metrics.